### PR TITLE
Add Netflow dashboards based on Logstash Netflow

### PR DESCRIPTION
**Pull request by andrewkroh**
Convert Logstash dashboards to work on Filebeat netflow data.

Changes

- Change all UUIDs to not conflict with Logstash dashboards.
- Rename fields used in dashboards.
```
	// These fields don't exist in the same capacity in ECS so just use destination.
	"geoip.autonomous_system": "destination.as.organization.name",
	"geoip.city_name":         "destination.geo.city_name",
	"geoip.country_name":      "destination.geo.country_name",
	"geoip.location":          "destination.geo.location",

	"geoip_dst.autonomous_system": "destination.as.organization.name",
	"geoip_src.autonomous_system": "source.as.organization.name",
	"host":                        "agent.hostname",
	"netflow.bytes":               "network.bytes",
	"netflow.direction":           "network.direction",
	"netflow.dst_addr":            "destination.ip",
	"netflow.dst_port_name":       "destination.port",
	"netflow.flow_locality":       "flow.locality",
	"netflow.input_snmp":          "netflow.ingress_interface",
	"netflow.ip_version":          "network.type",
	"netflow.last_switched":       "event.end",
	"netflow.output_snmp":         "netflow.egress_interface",
	"netflow.packets":             "network.packets",
	"netflow.protocol_name":       "network.transport",
	"netflow.src_addr":            "source.ip",
	"netflow.src_port_name":       "source.port",
	"netflow.tcp_flags_label":     "netflow.tcp_control_bits",
	"netflow.tos":                 "netflow.ip_class_of_service",
	"netflow.version":             "netflow.exporter.version",
	"netflow.vlan":                "netflow.vlan_id",
```

- Change index pattern from netflow-* to filebeat-*
- Add "input.type: netflow" filter. Uses can remove this to view other
  flow data in dashboards (this mostly works fine b/c of ECS).
- Prepend [Filebeat Netflow] to dashboards. And append [Filebeat Netflow] to visualizations.
- Update netflow pipeline to enrich flows with ASN info.

overview
conv-partners
traffic-analysis
top-n
geo
raw-flows
flow-exporters

### CODE REVIEW

File: x-pack/filebeat/magefile.go [hunk 1]
```diff
@@ -100,8 +105,8 @@ func fieldsYML() error {
 	return devtools.GenerateFieldsYAML(devtools.OSSBeatDir("module"), "module", "input")
 }
 
-// Dashboards collects all the dashboards and generates index patterns.
-func Dashboards() error {
```
Comment (jsoriano) [inline x-pack/filebeat/magefile.go]: Why removing this mage target? Should we do the change also in other beats?

File: x-pack/filebeat/module/netflow/_meta/kibana/7/dashboard/filebeat-netflow-autonomous-systems.json [hunk 2]
```diff
@@ -0,0 +1,597 @@
...
+            "col": 9,
+            "panelIndex": 8,
+            "panelRefName": "panel_7",
+            "row": 2,
+            "size_x": 4,
+            "size_y": 2
+          }
+        ],
+        "timeRestore": false,
+        "title": "[Filebeat] Netflow: Autonomous Systems",
```
Comment (jsoriano) [inline x-pack/filebeat/module/netflow/_meta/kibana/7/dashboard/filebeat-netflow-autonomous-systems.json]: It should be `[Filebeat Netflow] Autonomous Systems` (same for other dashboards)

File: x-pack/filebeat/module/netflow/_meta/kibana/7/dashboard/filebeat-netflow-autonomous-systems.json [hunk 3]
```diff
@@ -0,0 +1,597 @@
...
+            "filter": [],
+            "query": {
+              "query_string": {
+                "analyze_wildcard": true,
+                "query": "*"
+              }
+            }
+          }
+        },
+        "title": "Netflow: Dashboard Navigation [Filebeat]",
```
Comment (jsoriano) [inline x-pack/filebeat/module/netflow/_meta/kibana/7/dashboard/filebeat-netflow-autonomous-systems.json]: It should be `Dashboard Navigation [Filebeat Netflow]` (same for other visualizations)

File: x-pack/filebeat/module/netflow/log/ingest/pipeline.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        target_field: source.as.organization.name
+        ignore_missing: true
+  - rename:
+        field: destination.as.asn
+        target_field: destination.as.number
+        ignore_missing: true
+  - rename:
+        field: destination.as.organization_name
+        target_field: destination.as.organization.name
+        ignore_missing: true
```
Comment (jsoriano) [inline x-pack/filebeat/module/netflow/log/ingest/pipeline.yml]: Should these renames be added to breaking changes in changelog?

File: x-pack/filebeat/magefile.go (hunk 1 above)
Comment (andrewkroh) [reply x-pack/filebeat/magefile.go]: I do think we should make a similar change to all Beats to reuse the importable targets from dev-tools/mage/targets, but I'm not going to do this now. In fact I didn't mean to commit most of these changes. I was hacking at this to be able to import/export to Elastic Cloud.

I'm going to undo these and also fix the titles as you suggested.

File: x-pack/filebeat/module/netflow/log/ingest/pipeline.yml (hunk 4 above)
Comment (andrewkroh) [reply x-pack/filebeat/module/netflow/log/ingest/pipeline.yml]: No, those are new fields being added a few lines up. These rename are necessary to comply with ECS.

File: x-pack/filebeat/magefile.go (hunk 1 above)
Comment (andrewkroh) [reply x-pack/filebeat/magefile.go]: I left the changes to add KIBANA_URL support the export dashboard target, but I removed the refactoring of x-pack/filebeat/magefile.go.

File: x-pack/filebeat/module/netflow/log/ingest/pipeline.yml (hunk 4 above)
Comment (jsoriano) [reply x-pack/filebeat/module/netflow/log/ingest/pipeline.yml]: Oh, I see :+1:

**Pull request merged**